MainForm.cs

```diff
@@ -0,0 +1,249 @@
+                    if (nom > 7 || nom < 1)
+                    {
+                        textBox.ForeColor = Color.Red;
+                        textBox.Text = "\nError!";
+                        isDataCorrect = false;
+                    }
+                    else
+                    {
+                        nomOfCards.Add(nom);
+                        Colors color;
+                        char c = comb[i][colorPosition];
+                        if (!Enum.TryParse(Convert.ToString(c), out color))
+                        {
+                            Console.WriteLine($"Invalid value: {c}");
+                            return;
+                        }
+                        colors.Add(GetDisplayName(color));
+                        priotityOfColors.Add((int)color);
+                    }
+                }
+            }
+        }
+        public void findHighCard(ref int highCard, ref int n_highCard, int combLen, List<int> nomOfCards, List<string> colors, List<int> priotityOfColors)
+        {
+            for (int i = 1; i < combLen; i++)
+            {
+                if (nomOfCards[i] > highCard)
+                {
+                    highCard = nomOfCards[i];
+                    n_highCard = i;
+                }
+                else if (nomOfCards[i] == highCard)
+                {
+                    Colors color;
+                    int ind1 = priotityOfColors[i];
+                    int ind2 = priotityOfColors[n_highCard];
+                    n_highCard = ind1 < ind2 ? i : n_highCard;
+                    highCard = nomOfCards[n_highCard];
+                }
+            }
+        }
+        public void mainFunc(int combLen1, int combLen2)
+        {
+            Label[] labelsOnCards1 = { c11, c12, c13, c14, c15, c16, c17 };
+            Label[] labelsOnCards2 = { c21, c22, c23, c24, c25, c26, c27 };
+            Panel[] cardPanels1 = { card11, card12, card13, card14, card15, card16, card17 };
+            Panel[] cardPanels2 = { card21, card22, card23, card24, card25, card26, card27 };
+            List<int> nomOfCards1 = new List<int>();
+            List<int> nomOfCards2 = new List<int>();
+            List<string> colors1 = new List<string>();
+            List<string> colors2 = new List<string>();
+            List<int> priorityOfColors1 = new List<int>();
+            List<int> priorityOfColors2 = new List<int>();
+            string[] comb1 = new string[combLen1];
+            string[] comb2 = new string[combLen2];
+            int highCard1, n_highCard1, highCard2, n_highCard2, numOfWinner, nominalOfWinner;
+            string colorOfWinner;
+
+            for (int i = 0; i < combLen1; i++)
+                comb1[i] = textBox1.Lines[i + 1];
+            for (int i = 0; i < combLen2; i++)
+                comb2[i] = textBox2.Lines[i + 1];
+
+            bool isDataCorrect = true;
+            readData(ref isDataCorrect, combLen1, comb1, nomOfCards1, textBox1, colors1, priorityOfColors1);
```

Comment:
Why do you use the `ref` parameter here? You can return this value from `readData` method as a response value.